homeassistant/components/varta_storage/__init__.py

```diff
@@ -0,0 +1,203 @@
+            "state": result.from_grid_power,
+            "device_class": "power",
+            "state_class": "measurement",
+            "unit_of_measurement": "W",
+        }
+        thisdict[4] = {
+            "name": "VARTA state",
+            "state": result.state,
+            "device_class": "None",
+            "state_class": "None",
+            "unit_of_measurement": "",
+        }
+        thisdict[5] = {
+            "name": "VARTA state text",
+            "state": result.state_text,
+            "device_class": "None",
+            "state_class": "None",
+            "unit_of_measurement": "",
+        }
+        thisdict[6] = {
+            "name": "VARTA active power",
+            "state": result.active_power,
+            "device_class": "power",
+            "state_class": "measurement",
+            "unit_of_measurement": "W",
+        }
+        thisdict[7] = {
+            "name": "VARTA apparent power",
+            "state": result.apparent_power,
+            "device_class": "power",
+            "state_class": "measurement",
+            "unit_of_measurement": "W",
+        }
+        thisdict[8] = {
+            "name": "VARTA charge power",
+            "state": result.charge_power,
+            "device_class": "power",
+            "state_class": "measurement",
+            "unit_of_measurement": "W",
+        }
+        thisdict[9] = {
+            "name": "VARTA discharge power",
+            "state": result.discharge_power,
+            "device_class": "power",
+            "state_class": "measurement",
+            "unit_of_measurement": "W",
+        }
+        thisdict[10] = {
+            "name": "VARTA error code",
+            "state": result.error_code,
+            "device_class": "None",
+            "state_class": "None",
+            "unit_of_measurement": "",
+        }
+        thisdict[11] = {
+            "name": "VARTA production power",
+            "state": result.production_power,
+            "device_class": "power",
+            "state_class": "measurement",
+            "unit_of_measurement": "W",
+        }
+        thisdict[12] = {
+            "name": "VARTA total production power",
+            "state": result.total_production_power,
+            "device_class": "energy",
+            "state_class": "total_increasing",
+            "unit_of_measurement": "kWh",
+        }
+        thisdict[13] = {
+            "name": "VARTA total charged energy",
+            "state": result.total_charged_energy,
+            "device_class": "energy",
+            "state_class": "total",
+            "unit_of_measurement": "kWh",
+        }
+
+        return thisdict
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=1),
```

Comment:
Well. I absolutely understand your suggestion. On device-side there is a limit to 1 request per second - that's why i used this value.

Personally i would prefer to get frequent updates because especially the grid_power values are important to be updated high frequently if long-term measurement will be made with it. 

However, your suggested implementation of the SignificantlyChangedChecker is a very good idea. 
I tried to implement it according to the documentaiton by using

`python3 -m script.scaffold significant_change`

Afterwards a platform and test-file has been created. (you can see it in the current commit):
However, this part of the code never get called. Even if i modify the significant_change.py with a simple "Return False" every change will made to the sensor entity. 

I also set a breakpoint in the code and saw it will never called. Do you have a idea what would be needed to let the significant_change.py doing its job?

Reply:
Unfortunately I only read about this feature but have not yet tried to implement it. I will try to take a look to see whether I can figure out where the SignificantlyChangedChecker should be plugged but if someone more experimented with HA that I am already knows the answer, that would be nice ;)

> Personally i would prefer to get frequent updates because especially the grid_power values are important to be updated high frequently if long-term measurement will be made with it.

If the idea is to have a long-term measure of energy produced/stored/used, you should rely on `total_increasing` sensors rather than on `measurement` sensors, this way you do not need for frequently measured data.

Reply:
Okay, would be great. For the moment i will remove the signficant_change.py files from the commit.
I also can create a thread in the dev-forum and ask for support there

> If the idea is to have a long-term measure of energy produced/stored/used, you should rely on total_increasing sensors rather than on measurement sensors, this way you do not need for frequently measured data.

Yes - i am aware about that. Unfortunately the device itself does not provide a total value for the grid power measurement. It's only providing the current value. That's why i have to choose the measurement state_class.

However, the idea is: If a user would like to have long-term statistics (e.g. to use the grid power values provided by the integration within the HA integrated Energypanel) then the "[integration integration](https://www.home-assistant.io/integrations/integration/)" can be used to generate total_increasing sensors from that. According to the documentation for this integration i saw that as more frequent the values get updated, as more accurate the stats will get. Thats another reason why i choosed 1 update per second

Reply:
https://community.home-assistant.io/t/how-to-implement-significant-change-support-for-entity-integrations/380470

Reply:
Seems like the intention for this significant_change pull request (https://github.com/home-assistant/core/pull/45555) was not to filter updates to the state_machine. Its more to allow in the future to limit any external api calls / service calls. So i think thats not needed in this case